packages/provider-utils/src/convert-audio-input.ts

```diff
@@ -0,0 +1,218 @@
+import { TypeValidationError } from '@ai-sdk/provider';
+
+/**
+ * Converts various audio input formats to a standardized interface
+ * that provides methods to convert to different audio representations.
+ *
+ * @param audio - The audio input in various formats
+ * @returns An object with methods to convert to different audio formats
+ */
+export function convertAudioInput(
```

Comment:
Let's talk about this file in a chat - we have existing patterns and helpers that should be reused instead.